Add full ISO 8601 support to the en_CA ICU locale

There is no need to localize the date format in a written context because ISO 8601 is the international standard that apply worldwide! Seems the Android/ICU developers have missed that.

https://xkcd.com/1179/

This patch adheres to ISO 8601 (based on the ABNF from RFC 3999). In case more user friendliness is needed, the suggestion from [`systemd.time(7)`](https://manpages.debian.org/stable/systemd/systemd.time.7.en.html) to include the weekday in front of the ISO 8601 timestamp is used. Example:

```
Fri 2012-11-23 23:02:15
```


Confirmed working on:

* Android 10
  Apps: DeskClock, Settings (Security patch level), Files app (file info)
  Widgets: DatePicker

TODO:

* Widgets: DatePicker (month "April 2020" should be "2020-04 (April)")

The binaries files are not included but they can be generated from this source patch. They are not provided because you should not trust/use some binaries some random person (in this case me) is providing. Refer to the docs of external/icu/ for building the locale files.

project external/icu/
diff --git a/external/icu/icu4c/source/data/locales/en_CA.txt b/external/icu/icu4c/source/data/locales/en_CA.txt
index df35aa752..a3d02611e 100644
--- a/external/icu/icu4c/source/data/locales/en_CA.txt
+++ b/external/icu/icu4c/source/data/locales/en_CA.txt
@@ -179,48 +179,35 @@ en_CA{
                 "h:mm:ss a z",
                 "h:mm:ss a",
                 "h:mm a",
-                "EEEE, MMMM d, y",
-                "MMMM d, y",
-                "MMM d, y",
+                "EEE y-MM-dd",
                 "y-MM-dd",
-                "{1}, {0}",
+                "y-MM-dd",
+                "y-MM-dd",
+                "{1} {0}",
                 "{1} 'at' {0}",
                 "{1} 'at' {0}",
-                "{1}, {0}",
-                "{1}, {0}",
+                "{1} {0}",
+                "{1} {0}",
             }
             availableFormats{
-                GyMMMEd{"E, MMM d, y G"}
-                GyMMMd{"MMM d, y G"}
-                MEd{"E, MM-dd"}
-                MMMEd{"E, MMM d"}
-                MMMMd{"MMMM d"}
+                GyMMMEd{"E y-MM-dd G"}
+                GyMMMd{"y-MM-dd G"}
+                MEd{"E y-MM-dd"}
+                MMMEd{"E y-MM-dd"}
+                MMMMd{"y-MM-dd (MMM)"}
                 MMMd{"MMM d"}
                 MMdd{"MM-dd"}
                 Md{"MM-dd"}
                 yM{"y-MM"}
-                yMEd{"E, y-MM-dd"}
-                yMMMEd{"E, MMM d, y"}
-                yMMMd{"MMM d, y"}
+                yMEd{"E y-MM-dd"}
+                yMMMEd{"E y-MM-dd"}
+                yMMMd{"y-MM-dd"}
                 yMd{"y-MM-dd"}
                 yw{
                     one{"'week' w 'of' Y"}
                     other{"'week' w 'of' Y"}
                 }
             }
-            dayNames{
-                format{
-                    abbreviated{
-                        "Sun.",
-                        "Mon.",
-                        "Tue.",
-                        "Wed.",
-                        "Thu.",
-                        "Fri.",
-                        "Sat.",
-                    }
-                }
-            }
             dayPeriod{
                 format{
                     narrow{
@@ -250,39 +237,39 @@ en_CA{
             }
             intervalFormats{
                 MEd{
-                    M{"E, MM-dd – E, MM-dd"}
-                    d{"E, MM-dd – E, MM-dd"}
+                    M{"E y-MM-dd – E y-MM-dd"}
+                    d{"E y-MM-dd – E y-MM-dd"}
                 }
                 MMMEd{
-                    M{"E, MMM d – E, MMM d"}
-                    d{"E, MMM d – E, MMM d"}
+                    M{"E y-MM-dd – E y-MM-dd"}
+                    d{"E y-MM-dd – E y-MM-dd"}
                 }
                 MMMd{
-                    M{"MMM d – MMM d"}
-                    d{"MMM d – d"}
+                    M{"y-MM-dd – y-MM-dd"}
+                    d{"y-MM-dd – y-MM-dd"}
                 }
                 Md{
-                    M{"MM-dd – MM-dd"}
-                    d{"MM-dd – MM-dd"}
+                    M{"y-MM-dd – y-MM-dd"}
+                    d{"y-MM-dd – y-MM-dd"}
                 }
                 yM{
                     M{"y-MM – y-MM"}
                     y{"y-MM – y-MM"}
                 }
                 yMEd{
-                    M{"E, y-MM-dd – E, y-MM-dd"}
-                    d{"E, y-MM-dd – E, y-MM-dd"}
-                    y{"E, y-MM-dd – E, y-MM-dd"}
+                    M{"E y-MM-dd – E y-MM-dd"}
+                    d{"E y-MM-dd – E y-MM-dd"}
+                    y{"E y-MM-dd – E y-MM-dd"}
                 }
                 yMMMEd{
-                    M{"E, MMM d – E, MMM d, y"}
-                    d{"E, MMM d – E, MMM d, y"}
-                    y{"E, MMM d, y – E, MMM d, y"}
+                    M{"E y-MM-dd – E y-MM-dd"}
+                    d{"E y-MM-dd – E y-MM-dd"}
+                    y{"E y-MM-dd – E y-MM-dd"}
                 }
                 yMMMd{
-                    M{"MMM d – MMM d, y"}
-                    d{"MMM d – d, y"}
-                    y{"MMM d, y – MMM d, y"}
+                    M{"y-MM-dd – y-MM-dd"}
+                    d{"y-MM-dd – y-MM-dd"}
+                    y{"y-MM-dd – y-MM-dd"}
                 }
                 yMd{
                     M{"y-MM-dd – y-MM-dd"}
@@ -293,18 +280,18 @@ en_CA{
             monthNames{
                 format{
                     abbreviated{
-                        "Jan.",
-                        "Feb.",
-                        "Mar.",
-                        "Apr.",
+                        "Jan",
+                        "Feb",
+                        "Mar",
+                        "Apr",
                         "May",
-                        "Jun.",
-                        "Jul.",
-                        "Aug.",
-                        "Sep.",
-                        "Oct.",
-                        "Nov.",
-                        "Dec.",
+                        "Jun",
+                        "Jul",
+                        "Aug",
+                        "Sep",
+                        "Oct",
+                        "Nov",
+                        "Dec",
                     }
                 }
             }
@@ -348,382 +335,4 @@ en_CA{
         rightwards_arrows{"rightward arrow"}
         upwards_arrows{"upward arrows"}
     }
-    fields{
-        dayperiod{
-            dn{"a.m./p.m."}
-        }
-        dayperiod-narrow{
-            dn{"a.m./p.m."}
-        }
-        dayperiod-short{
-            dn{"a.m./p.m."}
-        }
-        fri-narrow{
-            relative{
-                "-1"{"last Fr"}
-                "0"{"this Fr"}
-                "1"{"next Fr"}
-            }
-            relativeTime{
-                future{
-                    one{"in {0} Fr"}
-                    other{"in {0} Fr’s"}
-                }
-                past{
-                    one{"{0} Fr ago"}
-                    other{"{0} Fr’s ago"}
-                }
-            }
-        }
-        fri-short{
-            relativeTime{
-                future{
-                    one{"in {0} Fri."}
-                    other{"in {0} Fri’s."}
-                }
-                past{
-                    one{"{0} Fri. ago"}
-                    other{"{0} Fri’s. ago"}
-                }
-            }
-        }
-        hour-narrow{
-            dn{"hr."}
-            relativeTime{
-                future{
-                    one{"in {0} hr."}
-                    other{"in {0} hrs."}
-                }
-                past{
-                    one{"{0} hr. ago"}
-                    other{"{0} hrs. ago"}
-                }
-            }
-        }
-        hour-short{
-            dn{"hr."}
-            relativeTime{
-                future{
-                    one{"in {0} hr."}
-                    other{"in {0} hrs."}
-                }
-                past{
-                    one{"{0} hr. ago"}
-                    other{"{0} hrs. ago"}
-                }
-            }
-        }
-        minute-narrow{
-            dn{"min."}
-            relativeTime{
-                future{
-                    one{"in {0} min."}
-                    other{"in {0} mins."}
-                }
-                past{
-                    one{"{0} min. ago"}
-                    other{"{0} mins. ago"}
-                }
-            }
-        }
-        minute-short{
-            dn{"min."}
-            relativeTime{
-                future{
-                    one{"in {0} min."}
-                    other{"in {0} mins."}
-                }
-                past{
-                    one{"{0} min. ago"}
-                    other{"{0} mins. ago"}
-                }
-            }
-        }
-        mon-narrow{
-            relative{
-                "-1"{"last Mo"}
-                "0"{"this Mo"}
-                "1"{"next Mo"}
-            }
-            relativeTime{
-                future{
-                    one{"in {0} Mo"}
-                    other{"in {0} Mo’s"}
-                }
-                past{
-                    one{"{0} Mo ago"}
-                    other{"{0} Mo’s ago"}
-                }
-            }
-        }
-        mon-short{
-            relativeTime{
-                future{
-                    one{"in {0} Mon."}
-                    other{"in {0} Mon’s."}
-                }
-                past{
-                    one{"{0} Mon. ago"}
-                    other{"{0} Mon’s. ago"}
-                }
-            }
-        }
-        month-narrow{
-            dn{"mo."}
-            relativeTime{
-                future{
-                    one{"in {0} mo."}
-                    other{"in {0} mos."}
-                }
-                past{
-                    one{"{0} mo. ago"}
-                    other{"{0} mos. ago"}
-                }
-            }
-        }
-        month-short{
-            dn{"mo."}
-            relativeTime{
-                future{
-                    one{"in {0} mo."}
-                    other{"in {0} mos."}
-                }
-                past{
-                    one{"{0} mo. ago"}
-                    other{"{0} mos. ago"}
-                }
-            }
-        }
-        quarter-narrow{
-            dn{"qtr."}
-            relativeTime{
-                future{
-                    one{"in {0} qtr."}
-                    other{"in {0} qtrs."}
-                }
-                past{
-                    one{"{0} qtr. ago"}
-                    other{"{0} qtrs. ago"}
-                }
-            }
-        }
-        quarter-short{
-            dn{"qtr."}
-            relativeTime{
-                future{
-                    one{"in {0} qtr."}
-                    other{"in {0} qtrs."}
-                }
-                past{
-                    one{"{0} qtr. ago"}
-                    other{"{0} qtrs. ago"}
-                }
-            }
-        }
-        sat-narrow{
-            relativeTime{
-                future{
-                    one{"in {0} Sa"}
-                    other{"in {0} Sa’s"}
-                }
-                past{
-                    one{"{0} Sa ago"}
-                    other{"{0} Sa’s ago"}
-                }
-            }
-        }
-        sat-short{
-            relativeTime{
-                future{
-                    one{"in {0} Sat."}
-                    other{"in {0} Sat’s."}
-                }
-                past{
-                    one{"{0} Sat. ago"}
-                    other{"{0} Sat’s. ago"}
-                }
-            }
-        }
-        second-narrow{
-            dn{"sec."}
-            relativeTime{
-                future{
-                    one{"in {0} sec."}
-                    other{"in {0} secs."}
-                }
-                past{
-                    one{"{0} sec. ago"}
-                    other{"{0} secs. ago"}
-                }
-            }
-        }
-        second-short{
-            dn{"sec."}
-            relativeTime{
-                future{
-                    one{"in {0} sec."}
-                    other{"in {0} secs."}
-                }
-                past{
-                    one{"{0} sec. ago"}
-                    other{"{0} secs. ago"}
-                }
-            }
-        }
-        sun-narrow{
-            relativeTime{
-                future{
-                    one{"in {0} Su"}
-                    other{"in {0} Su’s"}
-                }
-                past{
-                    one{"{0} Su ago"}
-                    other{"{0} Su’s ago"}
-                }
-            }
-        }
-        sun-short{
-            relativeTime{
-                future{
-                    one{"in {0} Sun."}
-                    other{"in {0} Sun’s."}
-                }
-                past{
-                    one{"{0} Sun. ago"}
-                    other{"{0} Sun’s. ago"}
-                }
-            }
-        }
-        thu-narrow{
-            relativeTime{
-                future{
-                    one{"in {0} Th"}
-                    other{"in {0} Th’s"}
-                }
-                past{
-                    one{"{0} Th ago"}
-                    other{"{0} Th’s ago"}
-                }
-            }
-        }
-        thu-short{
-            relativeTime{
-                future{
-                    one{"in {0} Thu."}
-                    other{"in {0} Thu’s."}
-                }
-                past{
-                    one{"{0} Thu. ago"}
-                    other{"{0} Thu’s ago"}
-                }
-            }
-        }
-        tue-narrow{
-            relativeTime{
-                future{
-                    one{"in {0} Tu"}
-                    other{"in {0} Tu’s"}
-                }
-                past{
-                    one{"{0} Tu ago"}
-                    other{"{0} Tu’s ago"}
-                }
-            }
-        }
-        tue-short{
-            relativeTime{
-                future{
-                    one{"in {0} Tue."}
-                    other{"in {0} Tue’s."}
-                }
-                past{
-                    one{"{0} Tue. ago"}
-                    other{"{0} Tue’s. ago"}
-                }
-            }
-        }
-        wed-narrow{
-            relative{
-                "-1"{"last We"}
-                "0"{"this We"}
-                "1"{"next We"}
-            }
-            relativeTime{
-                future{
-                    one{"in {0} We"}
-                    other{"in {0} We’s"}
-                }
-                past{
-                    one{"{0} We ago"}
-                    other{"{0} We’s ago"}
-                }
-            }
-        }
-        wed-short{
-            relativeTime{
-                future{
-                    one{"in {0} Wed."}
-                    other{"in {0} Wed’s."}
-                }
-                past{
-                    one{"{0} Wed. ago"}
-                    other{"{0} Wed’s. ago"}
-                }
-            }
-        }
-        week-narrow{
-            dn{"wk."}
-            relativePeriod{"the wk. of {0}"}
-            relativeTime{
-                future{
-                    one{"in {0} wk."}
-                    other{"in {0} wks."}
-                }
-                past{
-                    one{"{0} wk. ago"}
-                    other{"{0} wks. ago"}
-                }
-            }
-        }
-        week-short{
-            dn{"wk."}
-            relativePeriod{"the wk. of {0}"}
-            relativeTime{
-                future{
-                    one{"in {0} wk."}
-                    other{"in {0} wks."}
-                }
-                past{
-                    one{"{0} wk. ago"}
-                    other{"{0} wks. ago"}
-                }
-            }
-        }
-        year-narrow{
-            dn{"yr."}
-            relativeTime{
-                future{
-                    one{"in {0} yr."}
-                    other{"in {0} yrs."}
-                }
-                past{
-                    one{"{0} yr. ago"}
-                    other{"{0} yrs. ago"}
-                }
-            }
-        }
-        year-short{
-            dn{"yr."}
-            relativeTime{
-                future{
-                    one{"in {0} yr."}
-                    other{"in {0} yrs."}
-                }
-                past{
-                    one{"{0} yr. ago"}
-                    other{"{0} yrs. ago"}
-                }
-            }
-        }
-    }
 }
